### PR TITLE
Modules

### PR DESCRIPTION
**Pull request by aguglie**
Added support for modules.
Currently supports:

- RadioModule: set frequency, modulation, packet len, etc.
- PacketModificationModule: Modify any received module.
- PacketFilterModule: Filter incoming packets using reg. expressions.
- PacketRepeaterModule: Repeat any packet.
- RollJamModule: Implements a Roll Jam attack.
- FrequencyScanner: Scans a given range and searches for used frequencies.

### CODE REVIEW

File: examples/RFQuack-esp32-CC1101-serial/src/main.cpp [hunk 1]
```diff
@@ -31,16 +24,38 @@
...
- #include "rfquack.h"
-
- RadioA radio = new Module(RFQUACK_RADIO_PIN_CS, RFQUACK_RADIO_PIN_GDO0, RFQUACK_RADIO_PIN_GDO1);
+#include <rfquack.h>
 
- void setup() {
-   rfquack_setup(radio);
- }
+void setup() {
+  rfquack_setup(radioA, radioB, radioC, radioD, radioE);
```
Comment (phretor) [inline examples/RFQuack-esp32-CC1101-serial/src/main.cpp]: Might be a very stupid question, but please be patient :-)

Remind me why we can't do:

`rfquack_setup(radioA, radioB, null, null, null)`

Comment (aguglie) [reply examples/RFQuack-esp32-CC1101-serial/src/main.cpp]: You are right, we can do that!

I thought it was easier to leave there all the 5 radios in order to ease the plug-in of a new radio module.
But, as you suggest, the common use case is 1/2 radios; it's misleading to have all of them there.

Please, check the new commit :smile:

Comment (aguglie) [reply examples/RFQuack-esp32-CC1101-serial/src/main.cpp]: .... the common use case **IS** 2 radios :sweat_smile:
Is it better to default to two radios (radioA, radioB)? Should I add a  third one (radioC) commented?

**Review by phretor — APPROVED** [review]

File: examples/RFQuack-esp32-CC1101-serial/src/main.cpp (hunk 1 above)
Comment (phretor) [reply examples/RFQuack-esp32-CC1101-serial/src/main.cpp]: It doesn't really matter. You can mix and match. As long as the users understand that they have the "liberty" to choose, it's fine to me.

**Pull request merged**